addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,373 @@
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.sensebox.config.SenseBoxConfiguration;
+import org.openhab.binding.sensebox.internal.ExpiringCache;
+import org.openhab.binding.sensebox.model.SenseBoxData;
+import org.openhab.binding.sensebox.model.SenseBoxDescriptor;
+import org.openhab.binding.sensebox.model.SenseBoxLoc;
+import org.openhab.binding.sensebox.model.SenseBoxLocation;
+import org.openhab.binding.sensebox.model.SenseBoxSensor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link SenseBoxHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hakan Tandogan - Initial contribution
+ */
+
+public class SenseBoxHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(SenseBoxHandler.class);
+
+    protected SenseBoxConfiguration thingConfiguration;
+
+    private Gson gson = new Gson();
+
+    private SenseBoxData data = new SenseBoxData();
+
+    ScheduledFuture<?> refreshJob;
+
+    private final int CACHE_EXPIRY = 10 * 1000; // 10s
+
+    public SenseBoxHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+
+        thingConfiguration = getConfigAs(SenseBoxConfiguration.class);
+        String thingUid = getThing().getUID().toString();
+        thingConfiguration.setThingUid(thingUid);
+
+        logger.debug("Thing Configuration {} initialized {}", thingConfiguration.getThingUid(),
+                thingConfiguration.getSenseBoxId());
+
+        String offlineReason = "";
+        boolean validConfig = true;
+
+        if (StringUtils.trimToNull(thingConfiguration.getSenseBoxId()) == null) {
+            offlineReason = "senseBox ID is mandatory and must be configured";
+            logger.error("{}, disabling thing '{}'", offlineReason, thingUid);
+            validConfig = false;
+        }
+
+        if (thingConfiguration.getRefreshInterval() < MINIMUM_UPDATE_INTERVAL) {
+            logger.info("Refresh interval is much too small, setting to default of {} seconds",
+                    MINIMUM_UPDATE_INTERVAL);
+            thingConfiguration.setRefreshInterval(MINIMUM_UPDATE_INTERVAL);
+        }
+
+        if (validConfig) {
+            logger.debug("Thing Configuration: {}", thingConfiguration.toString());
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
No, but to be able to check for connectivity, I have to put the Thing into ONLINE mode, right? I would like to avoid expensive HTTP operations during initialization.